File: code/frontend/Tyr/src/model/Category.ts
import {SubCategory} from "./SubCategory";

/**
 * Category model
 *
 * @param id the id of the category
 * @param name the name of the category
 * @param description the description of the category
 * @param subCategories the subcategories of the category
 */
export interface Category {
    id?: number;
    name?: string;
    description?: string;
    subCategories?: SubCategory[];
}

File: code/backend/Odin/src/main/kotlin/pt/isel/odin/model/Category.kt
package pt.isel.odin.model

import com.fasterxml.jackson.annotation.JsonManagedReference
import jakarta.persistence.CascadeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.OneToMany
import jakarta.persistence.Table

@Entity
@Table(name = "categories")
class Category(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false)
    val name: String? = null,

    @OneToMany(mappedBy = "category", cascade = [CascadeType.ALL], fetch = FetchType.EAGER)
    @JsonManagedReference
    val subCategories: List<SubCategory>? = emptyList()
)
